src/state/opamSysInteract.ml

```diff
@@ -966,29 +965,90 @@ let packages_status ?(env=OpamVariable.Map.empty) config packages =
       |> package_set_of_pkgpath
     in
     compute_sets sys_installed
+  | Nix ->
+      (* We say all requested packages are available but uninstalled.
+         We could check that these packages are available in Nixpkgs,
+         but that would involve an expensive Nixpkgs evaluation.
+         Saying no packages are installed results in a warning that
+         conf packages depend on a 'system package that can no longer
+         be found.' But omitting them will mean that they won't be
+         added to the Nix derivation.
+      *)
+      let s_available = packages in
+      let s_not_found = OpamSysPkg.Set.empty in
+      let open OpamSysPkg in
+      { s_available; s_not_found }
   | Openbsd ->
     let sys_installed =
       run_query_command "pkg_info" ["-qP"]
       |> package_set_of_pkgpath
     in
     compute_sets sys_installed
 
+let stateless_install ?(env=OpamVariable.Map.empty) () =
+  match family ~env () with
+  | exception Failure _ -> false (* no depexts *)
+  | Nix -> false
```

Comment:
Shouldn't the function name be reversed? Nix is a stateless install, as well as no depext, but everything else should be stateful. So either the boolean needs to be flipped or the function needs to be renamed to `stateful_install`